src/main/java/uk/gov/companieshouse/efs/api/email/mapper/PaymentReportEmailMapper.java

```diff
@@ -39,11 +39,22 @@ public PaymentReportEmailMapper(final PaymentReportEmailConfig config, Identifie
 
     private PaymentReportEmailData fromPaymentReport(PaymentReportEmailModel model) {
         return PaymentReportEmailData.builder()
-            .withTo(model.getFileName().contains("Scottish") ? config.getScottishEmailAddress() : config.getFinanceEmailAddress())
+            .withTo(getEmailAddressForReportFilename(model))
             .withSubject(model.getFileName())
             .withFileLink(model.getFileLink())
             .withFileName(model.getFileName())
             .witHasNoPaymentTransactions(model.getHasNoPaymentTransactions())
             .build();
     }
+
+    private String getEmailAddressForReportFilename(final PaymentReportEmailModel model) {
+        String filename = model.getFileName();
+        if (filename.contains("Scottish")) {
+            return config.getScottishEmailAddress();
+        } else if (filename.contains("SH19")) {
```

Comment:
Please add unit test coverage for "SH19" case

Reply:
this took longer and the supporting methods now have a number of conditions to handle the expected data - would now be easier to add any further tests for new reports